apps/webhook/src/webhooks/usecases/execution-details/execution-details.usecase.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable, Module } from '@nestjs/common';
+import { ExecutionDetailsEntity, ExecutionDetailsRepository, MessageEntity } from '@novu/dal';
+import { ExecutionDetailsSourceEnum, ExecutionDetailsStatusEnum } from '@novu/shared';
+
+import { ExecutionDetailsCommand, WebhookCommand } from './execution-details.command';
+
+import { IWebhookResult } from '../../dtos/webhooks-response.dto';
+
+@Injectable()
+export class ExecutionDetails {
+  constructor(private executionDetailsRepository: ExecutionDetailsRepository) {}
+
+  async execute(command: ExecutionDetailsCommand): Promise<void> {
+    const executionDetailsEntity = this.mapWebhookEventIntoEntity(
+      command.webhook,
+      command.message,
+      command.webhookEvent
+    );
+
+    // TODO: Do we want to let this bubble up and break the webhook use case flow?
+    await this.executionDetailsRepository.create(executionDetailsEntity);
+  }
+
+  private mapWebhookEventIntoEntity(
+    webhook: WebhookCommand,
+    message: MessageEntity,
+    webhookEvent: IWebhookResult
+  ): ExecutionDetailsEntity {
+    const { environmentId: _environmentId, organizationId: _organizationId, providerId, type } = webhook;
+    const { _jobId, _messageTemplateId, _notificationId, _subscriberId, transactionId } = message;
+    const { externalId, attempts, response, row } = webhookEvent.event;
+
+    return {
+      _jobId,
+      _environmentId,
+      _organizationId,
+      _subscriberId,
+      _notificationId,
+      _notificationTemplateId: _messageTemplateId, // TODO: Confirm is this one.
+      providerId,
+      transactionId,
+      /**
+       * TODO: what should be the status? Mapper depending on Channel event status?
+       * e.g.: EmailEventStatusEnum, SmsEventStatusEnum, etc...
+       */
+      status: ExecutionDetailsStatusEnum.SUCCESS,
+      // TODO: Response brings parsed response by Novu, row is the raw details from provider
+      detail: response || row,
+      source: ExecutionDetailsSourceEnum.WEBHOOK,
+      // TODO: Webhook response might return the number of attempts. Should we use it as retry information?
+      isRetry: attempts && attempts > 0,
```

Comment:
I think not, we should keep the retry mechanics for our own execution flow of retrying a step. Just so we won't confuse the two.

Reply:
@davidsoderberg how do we identify in the webhook that the call is a retry? Also to identify it is a test?